core/dnsserver/address.go

```diff
@@ -94,3 +94,74 @@ const (
 	TransportTLS  = "tls"
 	TransportGRPC = "grpc"
 )
+
+const keyPartsSeparator = "##"
+
+type addrKey struct {
+	Transport string // dns, tls or grpc
+	Zone      string
+	Address   string
+	Port      string
+}
+
+func (k *addrKey) asKey() string {
+	return k.Transport + keyPartsSeparator + k.Zone + keyPartsSeparator + k.Address + keyPartsSeparator + k.Port
+}
+
+func parseAddrKey(key string) (*addrKey, error) {
+	keyParts := strings.Split(key, keyPartsSeparator)
+	if len(keyParts) != 4 {
+		return nil, fmt.Errorf("Provided value is not a key for addrKey, expect 4 parts joined by '%s' : %s", key, keyPartsSeparator)
+	}
+	return &addrKey{keyParts[0], keyParts[1], keyParts[2], keyParts[3]}, nil
+}
+
+func (k *addrKey) isMulticast() bool {
```

Comment:
Multicast? We want to care about this as well? Why? I thought this was going to be a purely textual change/check. We now also want to care about the type of addresses. 

NACK

at least in in this pr

Reply:
@miekg : what I name multicast is the usual key zone + port, when there is no address bound.

I guess we need to validate this overlapping use case:

not overlapping:
.:53{ bind ::1 ...}
.:53{ bind 127.0.0.1 ..}

overlapping
.:53{...}
.:53{ bind 127.0.0.1 ..}

I call ".:53" a "multicast" addrKey, but I can name it "unbound" .. 

What do you think.

NOTE: I do not introduce any new concept .. I only try to check the overlapping with existing concepts : having no bind directive, or having bind directive with one or several unicast addresses.